nck/utils/date_handler.py

```diff
@@ -38,13 +38,13 @@ def __get_previous_month_dates(current_date: date) -> Tuple[date, date]:
 
 
 def check_date_range_definition_conformity(start_date: date, end_date: date, date_range: str):
-    if start_date is None and end_date is None and date_range is None:
+    if (start_date is None and end_date is None) and date_range is None:
         raise Exception(
```

Comment:
It would be better to raise a custom exception (a class that inherits from the class `Exception`) that would be more specific and would give more info to the developer.